specs/configuration_files_specification.md

```diff
@@ -0,0 +1,79 @@
+which is readily available and maintained in all major distributions, rather than
+reimplementing this specification locally.
+
+This specification is agnostic toward the actual format and content of the configuration
+files, the precise path used under each top-level hierarchy, and also toward the
+filenames and extensions, with the exception of the `.d/` notation for drop-ins
+directories. It is strongly encouraged to enforce a specific suffix for the configuration
+files, in order to disambiguate (e.g.: with backup files), but the choice of which suffix
+to use is left to each implementation.
+
+## Storage Directories and Overrides
+In order to allow shipping system defaults owned by the OS vendor, while at the same
+time letting local users or admins override those defaults, `/usr/` and `/etc/` are both
+supported for storage of configuration files, with the latter having higher priority.
+The precise location under `/usr/` is left open for the implementation to decide - it
+could be hard-coded to `/usr/lib/` or it could be left to each application to pick from
+various options, such as `/usr/share/` or `/usr/etc/`.
+Programs must work correctly if no configuration files are found in `/etc/`.
+Optionally, `/run/` is also supported for ephemeral overrides.
+
+For example, `/usr/lib/foo/bar.conf` provides the default configuration file.
+If `/run/foo/bar.conf` is present and supported, it would take precedence
+over `/usr/lib/foo/bar.conf`.
+Finally, a user can create `/etc/foo/bar.conf` which would take precedence
+and completely override both.
+
+## Masking
+As a special override case, it must be possible to mask files across different
+locations by creating a symlink to `/dev/null` or an empty file.
+
+For example, an empty `/etc/foo/bar.conf` means that `/usr/foo/bar.conf` is
+masked and thus not parsed.
+
+## Drop-ins
+All configuration paths must support drop-ins. Supporting drop-ins means that in
+addition to parsing a full configuration file, an implementation also must parse all
+files in the drop-in directory associated with it, and merge the result in lexicographic
+order. Drop-ins are not full replacements, but must allow specifying a subset of settings
+present in the main configuration files.
+
+For example, in addition to parsing `/etc/foo/bar.conf`, `/etc/foo/bar.conf.d/a.conf` and
+`/etc/foo/bar.conf.d/b.conf` must also be parsed, in this order.
+
+Drop-ins always have higher precedence than the configuration file they refer to,
+regardless of under which hierarchy they are stored.
+
+For example, settings in `/usr/foo/bar.conf.d/a.conf` must override settings in
+`/etc/foo/bar.conf`.
+
+### Drop-ins without Main Configuration File
+Optionally, schemes with only drop-ins, without a 'main' configuration file, should also
+be supported by implementations. In such schemes many drop-ins are loaded from a common
+directory in each hierarchy.
+
+For example, `/usr/foo.d/a.conf`, `/usr/foo.d/b.conf` and `/etc/foo.d/c.conf` are all
```

Comment:
We use:
* `/usr/lib/foo/bar.conf.d`
* `/usr/foo/bar.conf.d`
* `/usr/foo.d/?.conf`
Could we please somehow unify the examples? That would make reading and understanding much easier.
I would suggest to use in all three cases `/usr/lib/foo/bar.conf.d`